File: algorithms/2_medium/P/00969_pancake_sorting/solution.cpp
typedef ptrdiff_t isize;

class Solution {
public:
    static vector<int> pancakeSort(vector<int>& arr) {
        const isize len = std::ssize(arr);
        auto res = vector<int>();

        for (isize i = len - 1; i >= 0; i -= 1) {
            isize idx_max = 0;
            for (isize j = 0; j <= i; j += 1) {
                if (arr[j] >= arr[idx_max]) {
                    idx_max = j;
                }
            }

            if (idx_max != i) {
                res.push_back(idx_max + 1);
                std::reverse(arr.begin(), arr.begin() + idx_max + 1);

                res.push_back(i + 1);
                std::reverse(arr.begin(), arr.begin() + i + 1);
            }
        }

        return res;
    }
};

File: algorithms/2_medium/P/00969_pancake_sorting/solution.rs
impl Solution {
    pub fn pancake_sort(mut arr: Vec<i32>) -> Vec<i32> {
        let len = arr.len();
        let mut res: Vec<i32> = Vec::new();

        for i in (0..len).rev() {
            let mut idx_max = 0;
            for j in 0..=i {
                if arr[j] >= arr[idx_max] {
                    idx_max = j;
                }
            }

            if idx_max != i {
                res.push(1 + idx_max as i32);
                arr[0..=idx_max].reverse();

                res.push(1 + i as i32);
                arr[0..=i].reverse();
            }
        }

        res
    }
}
